consensus/src/consensus/inner/commit.rs

```diff
@@ -239,7 +239,7 @@ impl ConsensusInner {
         }
 
         let digest = if self.ledger.requires_async_task(commitments.len(), serial_numbers.len()) {
-            let mut ledger = std::mem::replace(&mut self.ledger, DynLedger(Box::new(DummyLedger)));
+            let mut ledger = std::mem::replace(&mut self.ledger, DynLedger(Arc::new(DummyLedger)));
```

Comment:
i don't like changing this from a `Box` to an `Arc`. can we maybe use crossbeam scoped tasks?

Reply:
perhaps'; I'll look into it :+1:

Reply:
this might be impossible: https://users.rust-lang.org/t/async-spawn-blocking-but-scoped/38561/3; the alternatives I can see right now are:

- a `Mutex` over/inside `DynLedger`
- a hand-written `Clone` impl for `DynLedger` that checks that there is at most a single strong reference at the moment of cloning or it panics